### DISCUSSION TITLE
Run file button/command?

### DISCUSSION THREAD

**Discussion by jonastuttle** [zed-industries/zed#8294] [Feature Requests]
Loving the editor so far!

But I seem to only be able to run files through the command line; is there any support for a run button/command for my files?

**Comment by Moshyfawn** [comment]
If I understand correctly what you mean by "run files", I suggest following #7108 for information about the feature. It's called "runnables" and is similar to VSCode tasks. Although the UI is not yet available, [the initial version has recently been released](https://arc.net/l/quote/ljfvxkmq) to Preview.

**Reply by jonastuttle** [reply]
thank you!

**Comment by vixalien** [comment]
Now that the debugger and tasks have been shipped, is there hope for a run button next?

**Reply by olejorgenb** [reply]
Yeah, while it seems to be possible to configure a shortcut [1] to run the *current file* it would be great if this was built-in :crossed_fingers: 

[1] https://www.reddit.com/r/ZedEditor/comments/1cxpb9o/how_to_run_code_file_in_zed_using_keyboard/

**Comment by olejorgenb** [comment]
Depending on the language there is a run button inside-the code: (it's also a code action for that line)

<img width="373" height="65" alt="image" src="https://github.com/user-attachments/assets/bf0323d4-5b72-4dd6-abcf-eb9f6cb1e863" />

**Reply by olejorgenb** [reply]
And this can be triggered from anywhere in the file using "editor: spawn nearest task" (`editor::SpawnNearestTask`) (With obvious caveats if the file have multiple such "play" buttons )
